include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -868,6 +868,25 @@ struct __write_to_idx_if
     Assign __assign;
 };
 
+template <typename Assign = oneapi::dpl::__internal::__pstl_assign>
+struct __write_to_idx_if_else
```

Comment:
Maybe we also do not need the default for `_Assign` here.

Reply:
Fixed.  In truth, we probably don't need to support an arbitrary assignment operator here, but this makes it more consistent and extension easier.